src/Factory/Api/Mastodon/Attachment.php

```diff
@@ -51,39 +51,63 @@ public function createFromUriId(int $uriId): array
 	{
 		$attachments = [];
 		foreach (Post\Media::getByURIId($uriId, [Post\Media::AUDIO, Post\Media::VIDEO, Post\Media::IMAGE]) as $attachment) {
-			$filetype = !empty($attachment['mimetype']) ? strtolower(substr($attachment['mimetype'], 0, strpos($attachment['mimetype'], '/'))) : '';
-
-			if (($filetype == 'audio') || ($attachment['type'] == Post\Media::AUDIO)) {
-				$type = 'audio';
-			} elseif (($filetype == 'video') || ($attachment['type'] == Post\Media::VIDEO)) {
-				$type = 'video';
-			} elseif ($attachment['mimetype'] == 'image/gif') {
-				$type = 'gifv';
-			} elseif (($filetype == 'image') || ($attachment['type'] == Post\Media::IMAGE)) {
-				$type = 'image';
-			} else {
-				$type = 'unknown';
-			}
+			$attachments[] = $this->createFromMediaArray($attachment);
+		}
+
+		return $attachments;
+	}
+
+	/**
+	 * @param int $id id of the media
+	 * @return array
+	 * @throws HTTPException\InternalServerErrorException
+	 */
+	public function createFromId(int $id): array
+	{
+		$attachment = Post\Media::getById($id);
+		if (empty($attachment)) {
+			return [];
+		}
+		return $this->createFromMediaArray($attachment);
+	}
 
-			$remote = $attachment['url'];
-			if ($type == 'image') {
-				$url     = Post\Media::getPreviewUrlForId($attachment['id']);
+	/**
+	 * @param array $attachment
+	 * @return array
+	 * @throws HTTPException\InternalServerErrorException
+	 */
+	private function createFromMediaArray(array $attachment): array
+	{
+		$filetype = !empty($attachment['mimetype']) ? strtolower(substr($attachment['mimetype'], 0, strpos($attachment['mimetype'], '/'))) : '';
+
+		if (($filetype == 'audio') || ($attachment['type'] == Post\Media::AUDIO)) {
+			$type = 'audio';
+		} elseif (($filetype == 'video') || ($attachment['type'] == Post\Media::VIDEO)) {
+			$type = 'video';
+		} elseif ($attachment['mimetype'] == 'image/gif') {
+			$type = 'gifv';
+		} elseif (($filetype == 'image') || ($attachment['type'] == Post\Media::IMAGE)) {
+			$type = 'image';
+		} else {
+			$type = 'unknown';
+		}
+
+		$remote = $attachment['url'];
+		if ($type == 'image') {
+			$url     = Post\Media::getPreviewUrlForId($attachment['id']);
+			$preview = Post\Media::getPreviewUrlForId($attachment['id'], Proxy::SIZE_SMALL);
+		} else {
+			$url = $attachment['url'];
+
+			if (!empty($attachment['preview'])) {
 				$preview = Post\Media::getPreviewUrlForId($attachment['id'], Proxy::SIZE_SMALL);
 			} else {
-				$url = $attachment['url'];
-
-				if (!empty($attachment['preview'])) {
-					$preview = Post\Media::getPreviewUrlForId($attachment['id'], Proxy::SIZE_SMALL);
-				} else {
-					$preview = '';
-				}
+				$preview = '';
 			}
-
-			$object        = new \Friendica\Object\Api\Mastodon\Attachment($attachment, $type, $url, $preview, $remote);
-			$attachments[] = $object->toArray();
 		}
 
-		return $attachments;
+		$object        = new \Friendica\Object\Api\Mastodon\Attachment($attachment, $type, $url, $preview, $remote);
+		return $object->toArray();
```

Comment:
No need to cast to array at this point, it will be done automagically when trying to render the Status object to JSON.
```suggestion
		return new \Friendica\Object\Api\Mastodon\Attachment($attachment, $type, $url, $preview, $remote);
```

Reply:
Would it still internally call `toArray`? This here needs to be called as well: https://github.com/friendica/friendica/blob/develop/src/Object/Api/Mastodon/Attachment.php#L96-L105

Reply:
Yes, `json_encode` calls the `BaseDataTransferObject->toJSON` method recusively, which calls `toArray` internally.

Reply:
I just changed it, tried it out and received this:
`PHP Fatal error:  Uncaught TypeError: Friendica\Factory\Api\Mastodon\Attachment::createFromMediaArray(): Return value must be of type array, Friendica\Object\Api\Mastodon\Attachment returned in src/Factory/Api/Mastodon/Attachment.php:108`

Reply:
You need to change the return type-hint on the method declaration line.